common/assertion-functions/src/main/kotlin/com/testerum/common_assertion_functions/functions/builtin/ArrayFunctions.kt

```diff
@@ -0,0 +1,108 @@
+package com.testerum.common_assertion_functions.functions.builtin
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.ArrayNode
+import com.fasterxml.jackson.databind.node.NullNode
+import com.testerum.common_assertion_functions.functions.AssertionFailedException
+import com.testerum.common_assertion_functions.functions.AssertionFunction
+import java.math.BigDecimal
+
+object ArrayFunctions {
+
+    @AssertionFunction
+    fun isArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+    
+    @AssertionFunction
+    fun isArrayOrNull(actualNode: JsonNode) {
+        if (actualNode is NullNode) {
+            return
+        }
+
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isNonEmptyArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            val arrayNode = actualNode.get(0)
+            if (arrayNode != null) {
+                val actualSize = arrayNode.size()
+                if (actualSize > 0) {
+                    return
+                } else {
+                    throw AssertionFailedException("expected an non empty Array, but got an Array with ${actualSize} ${getElementsWord(actualSize)}: ${arrayNode.toString()}")
+                }
+            }
+        }
+
+        throw AssertionFailedException("expected an non empty Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isArrayWithSize(actualNode: JsonNode, expectedValue: BigDecimal) {
```

Comment:
not sure about this, but I got this idea after the other 2 comments below: what do you say about ``isArrayWithSizeExactly``?

Reply:
I like the other namings, but this "exactly" is noize word